### PR TITLE
feat: wipe student & login user from db

### PR DESCRIPTION
**Pull request by KoenDesplenter**
This pr creates the functionality to delete students or login user from the database and all their person related data.

### CODE REVIEW

**Comment by HuanYu-Tan** [comment]
There are still some bugs when deleting a user
`Invalid `prisma.session_keys.update()` invocation in
C:\Users\huany\les\SeLab2\OSOC-2\backend\orm_functions\session_key.ts:54:38

   51     newkey: string,
   52     date: Date
   53 ) {
→  54     return await prisma.session_keys.update(
  An operation failed because it depends on one or more records that were required but not fo
und. Record to update not found.
    at Object.request (C:\Users\huany\les\SeLab2\OSOC-2\backend\node_modules\@prisma\client\r
untime\index.js:45578:15)
    at async PrismaClient._request (C:\Users\huany\les\SeLab2\OSOC-2\backend\node_modules\@pr
isma\client\runtime\index.js:46405:18) {
  code: 'P2025',
  clientVersion: '3.12.0',
  meta: { cause: 'Record to update not found.' }
}
UNCAUGHT ERROR {"code":"P2025","clientVersion":"3.12.0","meta":{"cause":"Record to update not
 found."}}
` This one is from deleting yourself (you don't have a sessionkey then).

**Comment by HuanYu-Tan** [comment]
DELETE /admin/1
PrismaClientKnownRequestError: 
Invalid `prisma.login_user.delete()` invocation in
C:\Users\huany\les\SeLab2\OSOC-2\backend\orm_functions\login_user.ts:215:50

  212     },
  213 });
  214
→ 215 const person = await prisma.login_user.delete(
  Null constraint violation on the fields: (`login_user_id`)
    at Object.request (C:\Users\huany\les\SeLab2\OSOC-2\backend\node_modules\@prisma\client\r
untime\index.js:45578:15)
    at async PrismaClient._request (C:\Users\huany\les\SeLab2\OSOC-2\backend\node_modules\@pr
isma\client\runtime\index.js:46405:18) {
  code: 'P2011',
  clientVersion: '3.12.0',
  meta: { constraint: [ 'login_user_id' ] }
}
UNCAUGHT ERROR {"code":"P2011","clientVersion":"3.12.0","meta":{"constraint":["login_user_id"
]}}
This is when I try to delete Bob as OSOC

**Comment by jay-tux** [comment]
- fixed errors that were present -

**Review by Jonathan-Vanbrabant — CHANGES_REQUESTED** [review]
The osoc2 user can be deleted. Maybe we should make this user undeletable as main admin account.

**Comment by BramDevlaminck** [comment]
> The osoc2 user can be deleted. Maybe we should make this user undeletable as main admin account.

This will should be fixed once they pull from dev. I've done this but there are some issues left to fix. The "isAdmin" check and "isImmutable" checks seem to get the wrong ID, I'll try to take a look into fixing it tomorrow

**Review by BramDevlaminck — CHANGES_REQUESTED** [review]
We should pull from dev because there have been a lot of changes. i've done this and tried to fix the merge conflicts but some issues are still there bacause the wrong ID's get used. I'll take a look into fixing it tomorrow

**Review by BramDevlaminck — APPROVED** [review]
works for me!

File: backend/orm_functions/session_key.ts [hunk 1]
```diff
@@ -29,6 +29,7 @@ export async function addSessionKey(
  * @returns the loginUser associated with the key in a promise if the keys is valid otherwise an error in a promise
  */
 export async function checkSessionKey(key: string) {
+    console.log("key: " + key);
```
Comment (Mouwrice) [inline backend/orm_functions/session_key.ts:32]: The `console.log()` police is here! 🚨
Put down the log statement and come out with your hands above your head!

File: backend/routes/coach.ts [hunk 2]
```diff
@@ -110,40 +108,9 @@ export async function deleteCoach(
...
-                            })
-                            .catch(() =>
-                                Promise.reject(errors.cookServerError())
-                            );
-                    }
-                    return Promise.reject(errors.cookInvalidID());
-                });
+            return ormLU
+                .deleteLoginUserFromDB(parsed.data.id)
+                .then(() => Promise.resolve({}));
```
Comment (Mouwrice) [inline backend/routes/coach.ts:113]: Wouldn't an admin that is also a coach be able to delete himself?

**Review by Mouwrice — CHANGES_REQUESTED** [review]

File: backend/routes/coach.ts (hunk 2 above)
Comment (BramDevlaminck) [reply backend/routes/coach.ts:113]: nope! I've tested this and an admin that is also a coach can't delete himself.

**Review by Jonathan-Vanbrabant — APPROVED** [review]
Everything seems to work! Great work!

**Review by Mouwrice — APPROVED** [review]
Tip top!

**Pull request merged**